Feature/1378 fixed column table

JaydenIvanovic opened the pull request:
fixes #1378

JaydenIvanovic commented on src/components/Heroes/index.jsx:
```diff
@@ -100,7 +100,18 @@ class RequestLayer extends React.Component {
           />
-          <Table data={data} columns={_columns} loading={loading} />
+          <div style={{ position: 'relative' }}>
+            <Table data={data} columns={_columns} loading={loading} />
```
Planning to extract these two tables into a single component to hide the implementation details of the fixed column behaviour

JaydenIvanovic commented on src/components/Table/Table.jsx:
```diff
@@ -134,7 +134,7 @@ class Table extends React.Component {
     }
     return (
-      <StyledBody>
+      <StyledBody style={this.props.fixedStyle}>
```
Planning to make this prop a class name instead which can be generalised for other use cases

howardchung commented:
preview link:
https://deploy-preview-1410--opendota-stage.netlify.com/heroes/public

howardchung commented:
Would a similar approach work for freezing the header row?  That one may be trickier to implemnet.

GravityMsc commented on src/components/Heroes/index.jsx:
```diff
@@ -100,7 +100,13 @@ class RequestLayer extends React.Component {
-          <Table data={data} columns={_columns} loading={loading} />
+          <div style={{position:"relative"}}>
+            <Table data={data} columns={_columns} loading={loading} />
+            <Table data={data} columns={[_columns[0]]} loading={loading} fixedStyle={{
```
Well, I have a suggestion about `column`which is more  flexible. 
maybe you can add a prop  `fixed: right(or left)` in the column items, then you can fix some of columns right or left.

JaydenIvanovic replied:
I like this suggestion. Would the idea be that if multiple items had a prop of "fixed:right" that they would stack and be fixed to the right?

GravityMsc replied:
I think you could create a file of column tools, then you can filter some of columns use `Array.filter` or something else.
and if the filter column exists, such as 
```
const rightFixedColumns = columnTools.getRight(columns);
const leftFixedColumns = columnTools.getLeft(columns);
```
after that, use your code to create two fixed table above the original table. located right and left.

Sorry, I don't know whether you understand what I said. My English is not good.

JaydenIvanovic replied:
No worries, that explanation is perfect. I'll tackle this as soon as I work through the kinks of the fixed header.

JaydenIvanovic commented:
I believe a similar approach will work for the header row (seems to be implemented as such in the originally linked table). I will tackle that.

howardchung commented:
Seems to be a lint issue blocking the preview

howardchung commented:
There seem to be some rendering issues on mobile (testing with chrome emulator)

JaydenIvanovic commented:
I've fixed some of the mobile rendering issues.

There still seems to be one when scrolling vertically (especially when you hit the bottom of the "list") and another regarding the fixed hero column at a certain breakpoint (can see the hero column of the base table under it). 

Another known issue is that clicking column headers to sort the data won't sort the fixed hero's column (which I'm in the middle of fixing now). 

The header itself is now fixed correctly and aligned with the appropriate columns.

JaydenIvanovic commented:
Synced with master and finished off the table. I still need to do more testing to ensure that I haven't missed anything but I believe it's functioning as expected.
